src/modules/imageresizer/dll/ContextMenuHandler.cpp

```diff
@@ -260,17 +271,113 @@ HRESULT CContextMenuHandler::ResizePictures(CMINVOKECOMMANDINFO* pici)
+        DWORD fileCount = 0;
+        // Gets the list of files currently selected using the IShellItemArray
+        psiItemArray->GetCount(&fileCount);
+        // Iterate over the list of files
+        for (DWORD i = 0; i < fileCount; i++)
+        {
+            IShellItem* shellItem;
+            psiItemArray->GetItemAt(i, &shellItem);
+            LPWSTR itemName;
+            // Retrieves the entire file system path of the file from its shell item
+            shellItem->GetDisplayName(SIGDN_FILESYSPATH, &itemName);
+            CString fileName(itemName);
+            fileName.Append(_T("\r\n"));
+            // Write the file path into the input stream for image resizer
+            writePipe.Write(fileName, fileName.GetLength() * sizeof(TCHAR));
+        }
     }
 
     writePipe.Close();
 
     return S_OK;
 }
+
+HRESULT __stdcall CContextMenuHandler::GetTitle(IShellItemArray* /*psiItemArray*/, LPWSTR* ppszName)
+{
+    return SHStrDup(app_name.c_str(), ppszName);
+}
+
+HRESULT __stdcall CContextMenuHandler::GetIcon(IShellItemArray* /*psiItemArray*/, LPWSTR* ppszIcon)
+{
+    // Since ImageResizer is registered as a COM SurrogateServer the current module filename would be dllhost.exe. To get the icon we need the path of ImageResizerExt.dll, which can be obtained by passing the HINSTANCE of the dll
+    std::wstring iconResourcePath = get_module_filename(g_hInst_imageResizer);
+    iconResourcePath += L",-";
+    iconResourcePath += std::to_wstring(IDI_RESIZE_PICTURES);
+    return SHStrDup(iconResourcePath.c_str(), ppszIcon);
+}
+
+HRESULT __stdcall CContextMenuHandler::GetToolTip(IShellItemArray* /*psiItemArray*/, LPWSTR* ppszInfotip)
+{
+    *ppszInfotip = nullptr;
+    return E_NOTIMPL;
+}
+
+HRESULT __stdcall CContextMenuHandler::GetCanonicalName(GUID* pguidCommandName)
+{
+    *pguidCommandName = __uuidof(this);
+    return S_OK;
+}
+
+HRESULT __stdcall CContextMenuHandler::GetState(IShellItemArray* psiItemArray, BOOL fOkToBeSlow, EXPCMDSTATE* pCmdState)
+{
+    // Hide if the file is not an image
+    *pCmdState = ECS_HIDDEN;
+    // Suppressing C26812 warning as the issue is in the shtypes.h library
+#pragma warning(suppress : 26812)
+    PERCEIVED type;
+    PERCEIVEDFLAG flag;
+    IShellItem* shellItem;
+    //Check extension of first item in the list (the item which is right-clicked on)
+    psiItemArray->GetItemAt(0, &shellItem);
+    LPTSTR pszPath;
+    // Retrieves the entire file system path of the file from its shell item
+    shellItem->GetDisplayName(SIGDN_FILESYSPATH, &pszPath);
+    LPTSTR pszExt = PathFindExtension(pszPath);
+
+    // TODO: Instead, detect whether there's a WIC codec installed that can handle this file
+    AssocGetPerceivedType(pszExt, &type, &flag, NULL);
+
+    free(pszPath);
+    // If selected file is an image...
+    if (type == PERCEIVED_TYPE_IMAGE)
+    {
+        *pCmdState = CSettings::GetEnabled() ? ECS_ENABLED : ECS_HIDDEN;
```

Comment:
nit: we could check the value of `CSettings::GetEnabled()` and perform an early exit.